templates/vue/src/components/Lightbox.vue

```diff
@@ -0,0 +1,249 @@
+            @update-settings="updateH5pSettings"
+          />
+        </div>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script>
+import TextMedia from "./TextMedia"
+import VideoMedia from "./VideoMedia"
+import ExternalMedia from "./ExternalMedia"
+import H5PMedia from "./H5PMedia"
+import Helpers from "../utils/Helpers"
+
+export default {
+  name: "lightbox",
+  components: {
+    VideoMedia,
+    TextMedia,
+    ExternalMedia,
+    "h5p-media": H5PMedia,
+  },
+  props: {
+    nodeId: {
+      type: Number,
+      required: true,
+    },
+    tapestryApiClient: {
+      type: Object,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      node: {},
+      isLoaded: false,
+      dimensions: {
+        top: 100,
+        left: 50,
+      },
+      h5pSettings: {},
+    }
+  },
+  computed: {
+    lightboxContentStyles() {
+      return {
+        top: this.dimensions.top + "px",
+        left: this.dimensions.left + "px",
+        width: this.dimensions.width + "px",
+        height: this.dimensions.height + "px",
+      }
+    },
+    lightboxDimensions() {
+      const NORMAL_RADIUS = 140 // TODO: Refactor this to "constants" folder
+      if (!this.node) {
+        return {}
+      }
+
+      const { mediaWidth: width, mediaHeight: height } = this.node.typeData
+      const browserWidth = Helpers.getBrowserWidth()
+      const browserHeight = Helpers.getBrowserHeight()
+
+      let resizeRatio = 1
+      let videoWidth = width
+      let videoHeight = height
+
+      if (width > Helpers.getBrowserWidth()) {
+        resizeRatio *= browserWidth / width
+        videoWidth *= resizeRatio
+        videoHeight *= resizeRatio
+      }
+
+      if (videoHeight > browserHeight * resizeRatio) {
+        resizeRatio *= browserHeight / videoHeight
+        videoWidth *= resizeRatio
+        videoHeight *= resizeRatio
+      }
+
+      const nodeSpace = NORMAL_RADIUS * 2 * 1.3
+      const adjustedVideoHeight = Math.min(videoHeight, browserHeight - nodeSpace)
+      const adjustedVideoWidth = Math.min(videoWidth, browserWidth - nodeSpace)
+
+      const heightAdjustmentRatio = adjustedVideoHeight / videoHeight
+      const widthAdjustmentRatio = adjustedVideoWidth / videoWidth
+      let adjustmentRatio = widthAdjustmentRatio
+      let adjustedOn = "width"
+
+      if (Helpers.getAspectRatio() < 1) {
+        adjustedOn = "height"
+        adjustmentRatio = heightAdjustmentRatio
+      }
+
+      adjustmentRatio *= 0.95
+      return {
+        adjustedOn,
+        width: videoWidth * adjustmentRatio,
+        height: videoHeight * adjustmentRatio,
+      }
+    },
+  },
+  async mounted() {
+    const node = await this.tapestryApiClient.getNode(this.nodeId)
+    const meta = await this.tapestryApiClient.getNodeProgress(this.nodeId)
+    node.typeData.progress[0].value = meta.progress
+    node.typeData.progress[1].value = 1.0 - meta.progress
+
+    const settings = await this.tapestryApiClient.getH5pSettings()
```

Comment:
I believe this is a tapestry-wide setting that should be loaded at the beginning when the tapestry loads, so I don't think we should be calling it every time for a singe node as it will always give us the same data. This should be loaded when the tapestry is initially loaded and then made available to this component.